torch/csrc/jit/ir.cpp

```diff
@@ -251,6 +256,357 @@ std::ostream& operator<<(std::ostream & out, const Graph & g) {
   return out;
 }
 
+class PrettyPrintPass {
+  Graph& graph_;
+
+  // When printing a name if there is a conflict with an existing name in the
+  // graph, record the value -> new generated name mapping
+  std::unordered_map<const Value*, const Value*> aliases_;
+
+  // The Graph already tracks unique_names_, this is just for additional ones
+  // generated during printing
+  std::unordered_map<std::string, const Value*> generated_names_;
+
+  // Cache of value names
+  std::unordered_map<const Value*, std::string> value_names_;
+
+  template<class T>
+  void dualIterator(
```

Comment:
nit: this is often called `zipWith` (modulo the fact that we don't gather the returns).